mlir/lib/Dialect/Linalg/Transforms/TilingInterfaceImpl.cpp

```diff
@@ -324,7 +324,20 @@ struct LinalgOpTilingInterface
 // External Model for implementing `PartialReductionInterface` for `LinalgOp`s.
 //===----------------------------------------------------------------------===//
 
-/// External model implementation of PartialReductionInterface for LinalgOps.
+static AffineMap getPartialResultAffineMap(LinalgOp linalgOp,
+                                           ArrayRef<int> reductionDims,
+                                           unsigned resultNumber) {
+  AffineMap map =
+      linalgOp.getMatchingIndexingMap(linalgOp.getDpsInitOperand(resultNumber));
+  for (int redPos : reductionDims) {
+    map = map.insertResult(getAffineDimExpr(redPos, linalgOp.getContext()),
```

Comment:
This is assumuing that you can always add the reduction dimensions at the end?

Reply:
This is only for linalg ops. You can always add a reduction dimension at the end in a generic op, right?

Reply:
Also, this isn't really adding new dimensions to the operation's iteration space. This is only adding the reduction dimensions  reduction dimensions to the result/init (which did not have the iteration space's reduction dimensions before, because it was reduced).